### PR TITLE
Add cmap sub-command and update for latest Allsorts

### PR DESCRIPTION
**Pull request by wezm**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,6 +73,43 @@ The format is `{ppem_x}x{ppem_y}@{bit_depth}`, the files are named
 
     allsorts bitmaps -o noto-color-emoji NotoColorEmoji.ttf
 
+### `cmap`
+
+The `cmap` tool chooses an optimal `cmap` sub-table and dumps the character to
```
Comment (mikeday) [inline README.md]: Bit of a question there about what constitutes an optimal cmap table, basically the most complete / up to date Unicode one?

File: README.md [hunk 2]
```diff
@@ -72,6 +73,43 @@ The format is `{ppem_x}x{ppem_y}@{bit_depth}`, the files are named
...
+
+`-i`, `--index` is index of the font to dump (for TTC, WOFF2) (default: 0).
+
+#### Example
+
+    $ allsorts cmap --font profontn.otb
+    cmap sub-table encoding: Unicode
+    '' U+0000 -> 0
+    '' U+0001 -> 1
+    '' U+0002 -> 2
```
Comment (mikeday) [inline README.md:93]: If the single quotes are empty they could be omitted?

**Review by mikeday — APPROVED** [review]
Looks good!

File: README.md (hunk 1 above)
Comment (wezm) [reply README.md]: That's done by

https://github.com/yeslogic/allsorts/blob/aec512f9034de9e0a26e707e470cbc26917561b2/src/font.rs#L899

Perhaps "optimal" isn't the right word?

Comment (mikeday) [reply README.md]: I suppose it's "preferred".

**Comment by yisibl** [comment]
Hi @wezm Thank you for creating such an awesome tool.

I tested after running `cargo build` on this branch and it still seems to be missing OS/2 tables?

```bash
allsorts subset -t '床前明月光' SourceHanSerifCN/SourceHanSerifCN-Medium.ttf SourceHanSerifCN-Medium-subset.ttf
```

`ttx -l` diff tables name

image

**Comment by wezm** [comment]
Right yes, this work was focussed on `cmap`. I'll chat to the folks at work about this.

**Pull request merged**